hw8/task2.js

```diff
@@ -0,0 +1,314 @@
+}
+editEnterprise(1, 'Новое название предприятия 1');
+console.log(enterprises[0]);
+
+console.log('=======')
+
+/*
+6. Написать функцию для редактирования названия отдела. Принимает в качестве аргумента id отдела и новое имя отдела.
+
+Пример:
+editDepartment(7, "Новое название отдела")
+*/
+function editDepartment(deptId, newDeptName) {
+    const dept = getDepartmentById(deptId);
+    if (!dept) {
+        throw new Error("No such department");
+    }
+    dept.name = newDeptName;
+}
+
+function getDepartmentById(deptId) {
+    for (const company of enterprises) {
+        const foundDept = company.departments.find((dept) => dept?.id === deptId);
+        if (foundDept) {
+            return foundDept;
+        }
+    }
+    return null;
+};
+
+editDepartment(6, "Новое название отдела");
+console.log(enterprises[1]);
+
+console.log('=======')
+/*
+7. Написать функцию для удаления предприятия. В качестве аргумента принимает id предприятия.
+
+Пример:
+deleteEnterprise(1)
+*/
+
+function deleteEnterprise(companyId) {
+    const companyToDelete = getCompanyById(companyId);
+    enterprises.splice(companyToDelete, 1);
+}
+
+deleteEnterprise(1);
+console.log(enterprises);
+console.log('=======')
+/*
+8. Написать функцию для удаления отдела. В качестве аргумента принимает id отдела. Удалить отдел можно только, если в нем нет сотрудников.
+
+Пример:
+deleteDepartment(3)
+*/
+
+function deleteDepartment(deptId) {
+    const deptInQuestion = getDepartmentById(deptId);
+    if (!deptInQuestion) {
+        throw new Error("No such department");
+    }
+
+    if (deptInQuestion.employees_count > 0) {
+        throw new Error("Нельзя удалить отдел с сотрудниками!");
+    }
+
+    const company = getCompanyByDept(deptInQuestion);
+    const deptToDeleteIndex = company.departments.findIndex(dept => dept.id === deptId);
+    if (deptToDeleteIndex !== -1) {
+        company.departments.splice(deptToDeleteIndex, 1);
+    }
+}
+
+function getCompanyByDept(dept) {
+    return enterprises.find(company => company.departments.includes(dept));
+}
+
+deleteDepartment(10);
+console.log(enterprises);
+console.log('=======')
+
+/*
+9. Написать функцию для переноса сотрудников между отделами одного предприятия. В качестве аргумента принимает два значения: id отдела, из которого будут переноситься сотрудники и id отдела, в который будут переноситься сотрудники).
+
+Пример:
+moveEmployees(2, 3)
+*/
+
+function moveEmployees(iDdept1, iDdept2) {
+    const dept1 = getDepartmentById(iDdept1);
+    const dept2 = getDepartmentById(iDdept2);
+    console.log(dept1); // { id: 6, name: 'Новое название отдела', employees_count: 50 }
+    console.log(dept2); // { id: 7, name: 'Отдел маркетинга', employees_count: 20 }
+    
+    if (!dept1 || !dept2) {
+        throw new Error("Один из отделов не существует!");
+    }
+    
+    if (dept1.id === dept2.id) {
+        throw new Error("Выберите два разных отдела!");
+    }
+
+    const employeesToMove = dept1.employees_count;
```

Comment:
You've forgotten to check if this departments are in one enterprise. Just use getEnterprisesByDepartment method for both departments and compare the enterprise id